### PR TITLE
[SwiftActionsAPIGenerator] fixed undefined method `action_name'

### LINKED ISSUE #16: undefined method `action_name' for Fastlane::Actions::FirHelper:Class
`fastlane add_plugin fir_cli`

```
bundler: failed to load command: fastlane (/usr/local/Cellar/fastlane/2.145.0/libexec/bin/fastlane)
NoMethodError: [!] undefined method `action_name' for Fastlane::Actions::FirHelper:Class
  /usr/local/Cellar/fastlane/2.145.0/libexec/gems/fastlane-2.145.0/fastlane/lib/fastlane/swift_fastlane_api_generator.rb:82:in `block in initialize'
  /usr/local/Cellar/fastlane/2.145.0/libexec/gems/fastlane-2.145.0/fastlane/lib/fastlane/documentation/actions_list.rb:180:in `block in all_actions'
  /usr/local/Cellar/fastlane/2.145.0/libexec/gems/fastlane-2.145.0/fastlane/lib/fastlane/documentation/actions_list.rb:172:in `each'
  /usr/local/Cellar/fastlane/2.145.0/libexec/gems/fastlane-2.145.0/fastlane/lib/fastlane/documentation/actions_list.rb:172:in `all_actions'
  /usr/local/Cellar/fastlane/2.145.0/libexec/gems/fastlane-2.145.0/fastlane/lib/fastlane/swift_fastlane_api_generator.rb:81:in `initialize'
  /usr/local/Cellar/fastlane/2.145.0/libexec/gems/fastlane-2.145.0/fastlane/lib/fastlane/commands_generator.rb:342:in `new'
  /usr/local/Cellar/fastlane/2.145.0/libexec/gems/fastlane-2.145.0/fastlane/lib/fastlane/commands_generator.rb:342:in `block (2 levels) in run'
  /usr/local/Cellar/fastlane/2.145.0/libexec/gems/commander-fastlane-4.4.6/lib/commander/command.rb:178:in `call'
  /usr/local/Cellar/fastlane/2.145.0/libexec/gems/commander-fastlane-4.4.6/lib/commander/command.rb:153:in `run'
  /usr/local/Cellar/fastlane/2.145.0/libexec/gems/commander-fastlane-4.4.6/lib/commander/runner.rb:476:in `run_active_command'
  /usr/local/Cellar/fastlane/2.145.0/libexec/gems/fastlane-2.145.0/fastlane_core/lib/fastlane_core/ui/fastlane_runner.rb:76:in `run!'
  /usr/local/Cellar/fastlane/2.145.0/libexec/gems/commander-fastlane-4.4.6/lib/commander/delegates.rb:15:in `run!'
  /usr/local/Cellar/fastlane/2.145.0/libexec/gems/fastlane-2.145.0/fastlane/lib/fastlane/commands_generator.rb:349:in `run'
  /usr/local/Cellar/fastlane/2.145.0/libexec/gems/fastlane-2.145.0/fastlane/lib/fastlane/commands_generator.rb:41:in `start'
  /usr/local/Cellar/fastlane/2.145.0/libexec/gems/fastlane-2.145.0/fastlane/lib/fastlane/cli_tools_distributor.rb:119:in `take_off'
  /usr/local/Cellar/fastlane/2.145.0/libexec/gems/fastlane-2.145.0/bin/fastlane:23:in `<top (required)>'
  /usr/local/Cellar/fastlane/2.145.0/libexec/bin/fastlane:23:in `load'
  /usr/local/Cellar/fastlane/2.145.0/libexec/bin/fastlane:23:in `<top (required)>'
```

<details><summary>✅ fastlane environment ✅</summary>

### Stack

| Key                         | Value                                                   |
| --------------------------- | ------------------------------------------------------- |
| OS                          | 10.15.4                                                 |
| Ruby                        | 2.5.8                                                   |
| Bundler?                    | false                                                   |
| Git                         | git version 2.26.0                                      |
| Installation Source         | /usr/local/Cellar/fastlane/2.145.0/libexec/bin/fastlane |
| Host                        | Mac OS X 10.15.4 (19E266)                               |
| Ruby Lib Dir                | /usr/local/Cellar/ruby@2.5/2.5.8/lib                    |
| OpenSSL Version             | OpenSSL 1.1.1f  31 Mar 2020                             |
| Is contained                | false                                                   |
| Is homebrew                 | false                                                   |
| Is installed via Fabric.app | false                                                   |
| Xcode Path                  | /Applications/Xcode.app/Contents/Developer/             |
| Xcode Version               | 11.4.1                                                  |

### System Locale

| Variable | Value       |   |
| -------- | ----------- | - |
| LANG     | zh_CN.UTF-8 | ✅ |
| LC_ALL   |             |   |
| LANGUAGE |             |   |

### fastlane files:

<details><summary>`./fastlane/Fastfile.swift`</summary>

```ruby
// This file contains the fastlane.tools configuration
// You can find the documentation at https://docs.fastlane.tools
//
// For a list of all available actions, check out
//
//     https://docs.fastlane.tools/actions
//

import Foundation

class Fastfile: LaneFile {
	func debugLane() {
        desc("Push a new debug beta build to TestFlight")

        incrementBuildNumber()

	buildApp(workspace: "XXXX", scheme: "XXXX")

        uploadToTestflight(username: "XXXX", betaAppDescription: "XXXX", skipWaitingForBuildProcessing: true, teamId: "XXXX")

	}

}

```
</details>

**No Appfile found**

### fastlane gems

| Gem      | Version | Update-Status |
| -------- | ------- | ------------- |
| fastlane | 2.145.0 | ✅ Up-To-Date  |

### Loaded fastlane plugins:

| Plugin                  | Version | Update-Status |
| ----------------------- | ------- | ------------- |
| fastlane-plugin-fir_cli | 2.0.9   | ✅ Up-To-Date  |

<details><summary><b>Loaded gems</b></summary>

| Gem                       | Version      |
| ------------------------- | ------------ |
| slack-notifier            | 2.3.2        |
| rouge                     | 2.0.7        |
| xcpretty                  | 0.3.0        |
| terminal-notifier         | 2.0.0        |
| unicode-display_width     | 1.7.0        |
| terminal-table            | 1.8.0        |
| plist                     | 3.5.0        |
| CFPropertyList            | 3.0.2        |
| public_suffix             | 2.0.5        |
| addressable               | 2.7.0        |
| multipart-post            | 2.0.0        |
| word_wrap                 | 1.0.0        |
| tty-screen                | 0.7.1        |
| tty-cursor                | 0.7.1        |
| tty-spinner               | 0.9.3        |
| babosa                    | 1.0.3        |
| colored                   | 1.2          |
| highline                  | 1.7.10       |
| commander-fastlane        | 4.4.6        |
| excon                     | 0.73.0       |
| faraday                   | 0.17.3       |
| unf_ext                   | 0.0.7.7      |
| unf                       | 0.1.4        |
| domain_name               | 0.5.20190701 |
| http-cookie               | 1.0.3        |
| faraday-cookie_jar        | 0.0.6        |
| faraday_middleware        | 0.13.1       |
| fastimage                 | 2.1.7        |
| gh_inspector              | 1.1.3        |
| json                      | 2.3.0        |
| mini_magick               | 4.10.1       |
| multi_xml                 | 0.6.0        |
| rubyzip                   | 1.3.0        |
| security                  | 0.1.3        |
| xcpretty-travis-formatter | 1.0.0        |
| dotenv                    | 2.7.5        |
| bundler                   | 2.1.4        |
| naturally                 | 2.2.0        |
| simctl                    | 1.6.8        |
| jwt                       | 2.1.0        |
| uber                      | 0.1.0        |
| declarative               | 0.0.10       |
| declarative-option        | 0.1.0        |
| representable             | 3.0.4        |
| retriable                 | 3.1.2        |
| mini_mime                 | 1.0.2        |
| multi_json                | 1.14.1       |
| signet                    | 0.14.0       |
| httpclient                | 2.8.3        |
| google-api-client         | 0.36.4       |
| emoji_regex               | 1.0.1        |
| aws-sigv4                 | 1.1.1        |
| aws-sdk-kms               | 1.30.0       |
| aws-sdk-s3                | 1.61.2       |
| nanaimo                   | 0.2.6        |
| colored2                  | 3.1.2        |
| claide                    | 1.0.3        |
| atomos                    | 0.1.3        |
| xcodeproj                 | 1.16.0       |
| os                        | 1.1.0        |
| memoist                   | 0.16.2       |
| googleauth                | 0.12.0       |
| digest-crc                | 0.5.1        |
| google-cloud-errors       | 1.0.0        |
| google-cloud-env          | 1.3.1        |
| google-cloud-core         | 1.5.0        |
| google-cloud-storage      | 1.26.0       |
| aws-eventstream           | 1.1.0        |
| aws-partitions            | 1.297.0      |
| jmespath                  | 1.4.0        |
| aws-sdk-core              | 3.94.0       |
| http-accept               | 1.7.0        |
| mime-types-data           | 3.2019.1009  |
| mime-types                | 3.3.1        |
| netrc                     | 0.11.0       |
| rest-client               | 2.1.0        |
| api_tools                 | 0.1.0        |
| thor                      | 0.20.3       |
| admqr_knife               | 0.1.5        |
| ruby_android_apk          | 0.7.7.1      |
| chunky_png                | 1.3.11       |
| rqrcode                   | 0.10.1       |
| fir-cli                   | 2.0.9        |
| fastlane-plugin-fir_cli   | 2.0.9        |
</details>

*generated on:* **2020-04-17**
</details>

### PR DESCRIPTION
**Pull request by cdoky**
### Checklist
- [x] I've run `bundle exec rspec` from the root directory to see all new and existing tests pass
- [x] I've followed the _fastlane_ code style and run `bundle exec rubocop -a` to ensure the code style is valid
- [x] I've read the [Contribution Guidelines](https://github.com/fastlane/fastlane/blob/master/CONTRIBUTING.md)
- [x] I've updated the documentation if necessary.

### Motivation and Context

Resolves https://github.com/FIRHQ/fastlane-plugin-fir_cli/issues/16

### Testing Steps

1. add `"fastlane", git:"https://github.com/philCc/fastlane.git"` in your Gemfile
2. run shell cmd: `bundle update fastlane && fastlane update_plugins`

### CODE REVIEW

**Review by joshdholtz — COMMENTED** [review]
This fix seems legit but I'm very confused on how this error was even happening in the first place 😊 Do you happen to have a stack trace with this error? I'd like to see how this issue was happening!

**Comment by philCc** [comment]
Hi @joshdholtz 

Here is an issue, you can have a look: https://github.com/FIRHQ/fastlane-plugin-fir_cli/issues/16

**Comment by philCc** [comment]
If you use fastlane swift version and add plugin `fir_cli`, there will be happen a error:

`NoMethodError: [!] undefined method ‘action_name’ for Fastlane::Actions::FirHelper:Class `

**Comment by joshdholtz** [comment]
@philCc Ahhhh, I see! I think the fix should actually be made in the plugin 😊 

The issue is these classes within the action itself 👉  https://github.com/FIRHQ/fastlane-plugin-fir_cli/blob/master/lib/fastlane/plugin/fir_cli/actions/fir_cli_action.rb#L179-L200

1) You could fix this by putting a `private` above that section of code (I think)
2) Or these classes should get moved into https://github.com/FIRHQ/fastlane-plugin-fir_cli/blob/master/lib/fastlane/plugin/fir_cli/helper/fir_cli_helper.rb

Let me know your thoughts on that!

**Comment by philCc** [comment]
> @philCc Ahhhh, I see! I think the fix should actually be made in the plugin 😊
> 
> The issue is these classes within the action itself 👉 https://github.com/FIRHQ/fastlane-plugin-fir_cli/blob/master/lib/fastlane/plugin/fir_cli/actions/fir_cli_action.rb#L179-L200
> 
> 1. You could fix this by putting a `private` above that section of code (I think)
> 2. Or these classes should get moved into https://github.com/FIRHQ/fastlane-plugin-fir_cli/blob/master/lib/fastlane/plugin/fir_cli/helper/fir_cli_helper.rb
> 
> Let me know your thoughts on that!

@joshdholtz However, I think we should do a validation in the FastLane layer to prevent other plug-ins from experiencing the same problem

File: fastlane/lib/fastlane/swift_fastlane_api_generator.rb [hunk 1]
```diff
@@ -79,6 +79,11 @@ def initialize(target_output_path: "swift")
       available_external_actions = Fastlane.external_actions || []
       available_actions = []
       ActionsList.all_actions do |action|
+        next if (begin
+                   action.action_name
+                 rescue
+                   nil
+                 end).nil?
```
Comment (joshdholtz) [inline fastlane/lib/fastlane/swift_fastlane_api_generator.rb:86]:
```suggestion
        next unless action.response_to?(:name)
```

File: fastlane/lib/fastlane/swift_fastlane_api_generator.rb [hunk 2]
```diff
@@ -108,6 +113,11 @@ def initialize(target_output_path: "swift")
       # Excludes all actions that aren't pluign actions (including external actions)
       available_actions = []
       ActionsList.all_actions do |action|
+        next if (begin
+                   action.action_name
+                 rescue
+                   nil
+                 end).nil?
```
Comment (joshdholtz) [inline fastlane/lib/fastlane/swift_fastlane_api_generator.rb:120]:
```suggestion
        next unless action.response_to?(:name)
```

File: fastlane/lib/fastlane/swift_fastlane_api_generator.rb [hunk 3]
```diff
@@ -151,6 +161,11 @@ def generate_swift
 
       tool_details = []
       ActionsList.all_actions do |action|
+        next if (begin
+                   action.action_name
+                 rescue
+                   nil
+                 end).nil?
```
Comment (joshdholtz) [inline fastlane/lib/fastlane/swift_fastlane_api_generator.rb:168]:
```suggestion
        next unless action.response_to?(:name)
```

**Review by joshdholtz — CHANGES_REQUESTED** [review]
Good call on keeping this in here. How does something like this look though? I think this should still fix it for you!

File: fastlane/lib/fastlane/swift_fastlane_api_generator.rb (hunk 3 above)
Comment (philCc) [reply fastlane/lib/fastlane/swift_fastlane_api_generator.rb:168]: Sorry, an error will occur here
```
Updating plugin dependencies...
[✔] 🚀
/Users/pz/.rvm/gems/ruby-2.6.5/bundler/gems/fastlane-plugin-fir_cli-45c6ca10c187/lib/fastlane/plugin/fir_cli/actions/fir_cli_action.rb:183: warning: already initialized constant FIR::VERSION
/Users/pz/.rvm/gems/ruby-2.6.5/gems/fir-cli-2.0.10/lib/fir/version.rb:4: warning: previous definition of VERSION was here
+---------------------------+---------+--------------------------+
|                          Used plugins                          |
+---------------------------+---------+--------------------------+
| Plugin                    | Version | Action                   |
+---------------------------+---------+--------------------------+
| fastlane-plugin-pgyer     | 0.2.2   | pgyer                    |
| fastlane-plugin-sentry    | 1.7.0   | sentry_upload_dsym       |
|                           |         | sentry_set_commits       |
|                           |         | sentry_upload_sourcemap  |
|                           |         | sentry_finalize_release  |
|                           |         | sentry_upload_file       |
|                           |         | sentry_upload_proguard   |
|                           |         | sentry_create_release    |
| fastlane-plugin-fir_cli   | 2.0.9   | fir_cli                  |
| fastlane-plugin-ding_talk | 0.1.1   | ding_talk                |
+---------------------------+---------+--------------------------+

[09:28:39]: Unknown method 'response_to?'

[!] To call another action from an action use `other_action.response_to?` instead
```

**Comment by philCc** [comment]
> @philCc Ahhhh, I see! I think the fix should actually be made in the plugin 😊
> 
> The issue is these classes within the action itself 👉 https://github.com/FIRHQ/fastlane-plugin-fir_cli/blob/master/lib/fastlane/plugin/fir_cli/actions/fir_cli_action.rb#L179-L200
> 
> 1. You could fix this by putting a `private` above that section of code (I think)
> 2. Or these classes should get moved into https://github.com/FIRHQ/fastlane-plugin-fir_cli/blob/master/lib/fastlane/plugin/fir_cli/helper/fir_cli_helper.rb
> 
> Let me know your thoughts on that!

The solution is feasible, I will create a PR for this plugin.
But, I still think we should do a validation on the Fastlane layer

**Comment by philCc** [comment]
> Good call on keeping this in here. How does something like this look though? I think this should still fix it for you!

Hi, @joshdholtz 
An error will occur here
```
Updating plugin dependencies...
[✔] 🚀
/Users/pz/.rvm/gems/ruby-2.6.5/bundler/gems/fastlane-plugin-fir_cli-45c6ca10c187/lib/fastlane/plugin/fir_cli/actions/fir_cli_action.rb:183: warning: already initialized constant FIR::VERSION
/Users/pz/.rvm/gems/ruby-2.6.5/gems/fir-cli-2.0.10/lib/fir/version.rb:4: warning: previous definition of VERSION was here
+---------------------------+---------+--------------------------+
|                          Used plugins                          |
+---------------------------+---------+--------------------------+
| Plugin                    | Version | Action                   |
+---------------------------+---------+--------------------------+
| fastlane-plugin-pgyer     | 0.2.2   | pgyer                    |
| fastlane-plugin-sentry    | 1.7.0   | sentry_upload_dsym       |
|                           |         | sentry_set_commits       |
|                           |         | sentry_upload_sourcemap  |
|                           |         | sentry_finalize_release  |
|                           |         | sentry_upload_file       |
|                           |         | sentry_upload_proguard   |
|                           |         | sentry_create_release    |
| fastlane-plugin-fir_cli   | 2.0.9   | fir_cli                  |
| fastlane-plugin-ding_talk | 0.1.1   | ding_talk                |
+---------------------------+---------+--------------------------+

[09:28:39]: Unknown method 'response_to?'

[!] To call another action from an action use `other_action.response_to?` instead
```

**Comment by joshdholtz** [comment]
@philCc I totally did lol 😛 I typo-ed 🤷‍♂️

**Review by joshdholtz — APPROVED** [review]
This looks great! Thanks for the contribution ❤️

**Pull request merged**